### PR TITLE
fix: remove click in favor of onClick

### PR DESCRIPTION
**Pull request by beawar**
- [x] Update with Zextras/carbonio-design-system#172

Fix types.
Extract callbacks and memoize them.
Clean up the repo from the unused old files.

refs: SHELL-67

### CODE REVIEW

File: src/settings/components/settings-header.tsx [hunk 1]
```diff
@@ -26,25 +27,31 @@ type SettingsHeaderProps = {
...
-		},
-		{
-			id: 'general',
-			label: title,
-			click: (): void => history.push(`/${SETTINGS_APP_ID}/`)
-		}
-	];
-	const search = history.location?.search;
+	const params = useParams();
+	const crumbs = useMemo(
```
Comment (CataldoMazzilli) [inline src/settings/components/settings-header.tsx:34]: these crumbs, that are used for each module, do not work...we need to open an issue?

File: src/settings/components/settings-header.tsx [hunk 2]
```diff
@@ -53,7 +60,7 @@ const SettingsHeader: FC<SettingsHeaderProps> = ({ onSave, onCancel, isDirty, ti
 				1
 			);
 		}
-	}, [history, history.location, history.location.search, search, useparam]);
+	}, [history, history.location, history.location.search, search, params]);
 	return (
 		<>
 			<RouteLeavingGuard when={isDirty} onSave={onSave}>
```
Comment (CataldoMazzilli) [inline src/settings/components/settings-header.tsx:66]: Some times seems appears also when you do not edit anything... maybe we che investigate and align the RouteLeavingGuard with the Files one?

**Review by CataldoMazzilli — DISMISSED** [review]

File: src/settings/components/settings-header.tsx (hunk 2 above)
Comment (beawar) [reply src/settings/components/settings-header.tsx:66]: I think there is a more general problem with the settings section itself, which need to be cleared and refactored in a separated issue. We can of course open an issue to check this bug.

File: src/settings/components/settings-header.tsx (hunk 1 above)
Comment (beawar) [reply src/settings/components/settings-header.tsx:34]: Good catch, I didn't notice. Yep, we should, I'm on it.

File: src/settings/components/settings-header.tsx (hunk 2 above)
Comment (beawar) [reply src/settings/components/settings-header.tsx:66]: SHELL-80 opened for the unsaved changes modal opening when no change has been made

File: src/settings/components/settings-header.tsx (hunk 1 above)
Comment (beawar) [reply src/settings/components/settings-header.tsx:34]: SHELL-81 opened for the breadcrumb not working well

**Review by nubsthead — DISMISSED** [review]

**Review by nubsthead — APPROVED** [review]

**Review by CataldoMazzilli — DISMISSED** [review]

**Review by nubsthead — DISMISSED** [review]

**Review by CataldoMazzilli — DISMISSED** [review]

**Review by lisaparma — DISMISSED** [review]

**Review by nubsthead — DISMISSED** [review]

**Review by nubsthead — APPROVED** [review]

**Review by gnekoz — APPROVED** [review]

**Review by CataldoMazzilli — APPROVED** [review]

**Review by lisaparma — APPROVED** [review]

**Pull request merged**